src/main/kotlin/org/wycliffeassociates/otter/jvm/recorder/device/AudioRecorder.kt

```diff
@@ -0,0 +1,85 @@
+package org.wycliffeassociates.otter.jvm.recorder.device
+
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import org.wycliffeassociates.otter.common.device.IAudioRecorder
+import javax.sound.sampled.AudioFormat
+import javax.sound.sampled.AudioSystem
+
+class AudioRecorder : IAudioRecorder {
+
+    private val monitor = Object()
+
+    @Volatile
+    private var stop = false
+
+    companion object {
+        val SAMPLE_RATE = 44100F // Hz
+        val SAMPLE_SIZE = 16 // bits
+        val CHANNELS = 1
+        val SIGNED = true
+        val BIG_ENDIAN = false
+        val FORMAT = AudioFormat(
+            SAMPLE_RATE,
+            SAMPLE_SIZE,
+            CHANNELS,
+            SIGNED,
+            BIG_ENDIAN
+        )
+        val BUFFER_SIZE = 1024
+    }
+
+    private var line = AudioSystem.getTargetDataLine(FORMAT)
+    private val audioByteObservable = PublishSubject.create<ByteArray>()
+    private val recordingStream = Observable.fromCallable {
+        val byteArray = ByteArray(BUFFER_SIZE)
+        var totalRead = 0
+        while (true) {
+            if (line.isOpen || line.available() > 0) {
+                totalRead += line.read(byteArray, 0, byteArray.size)
+                audioByteObservable.onNext(byteArray)
+            } else {
+                try {
+                    synchronized(monitor) {
+                        monitor.wait()
+                    }
+                } catch (e: InterruptedException) {
+                    stop()
+                }
+            }
+            if (stop) {
+                line.close()
+                break
+            }
+        }
+    }.subscribeOn(Schedulers.io())
+        .subscribe()
+
+    @Synchronized //Synchronized so as to not subscribe to multiple streams on quick multipress
+    override fun start() {
+        line.open(FORMAT)
+        line.start()
+        synchronized(monitor) {
+            monitor.notify()
+        }
+    }
+
+    override fun pause() {
+        line.stop()
+        line.close()
+    }
```

Comment:
`stop()` sets a flag and waits for a current read to complete.  `pause()` does not and just slams the line closed.  I recommend adding a `pause` flag like the `stop` flag, and make the `pause()` implementation look a lot like `stop()`.

Then the line-closing section in the read loop will be like this:
```kotlin
if (stop || pause) {
    line.close()
    if (stop) break
}